src/WebJobs.Script.WebHost/Middleware/HostAvailabilityCheckMiddleware.cs

```diff
@@ -15,31 +15,37 @@ public partial class HostAvailabilityCheckMiddleware
         private readonly RequestDelegate _next;
         private readonly ILogger<HostAvailabilityCheckMiddleware> _logger;
         private readonly IOptionsMonitor<ScriptApplicationHostOptions> _applicationHostOptions;
+        private readonly IScriptHostManager _scriptHostManager;
 
-        public HostAvailabilityCheckMiddleware(RequestDelegate next, ILoggerFactory loggerFactory, IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions)
+        public HostAvailabilityCheckMiddleware(RequestDelegate next, ILoggerFactory loggerFactory, IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IScriptHostManager scriptHostManager)
         {
             _next = next;
             _logger = loggerFactory.CreateLogger<HostAvailabilityCheckMiddleware>();
             _applicationHostOptions = applicationHostOptions;
+            _scriptHostManager = scriptHostManager;
         }
 
-        public async Task Invoke(HttpContext httpContext, IScriptHostManager scriptHostManager)
```

Comment:
Our custom scoped DI resolution logic is expensive. Where we can, we should move this off of the hot path, and initialize once.